sweep/sweeper.go

```diff
@@ -979,6 +1001,37 @@ func (s *UtxoSweeper) waitForSpend(outpoint wire.OutPoint,
 	return spendEvent.Cancel, nil
 }
 
+// PendingInputs returns the set of inputs that the UtxoSweeper is currently
+// attempting to sweep.
+func (s *UtxoSweeper) PendingInputs() ([]*PendingInput, error) {
+	respChan := make(chan []*PendingInput, 1)
+	select {
+	case s.pendingSweepsReqs <- &pendingSweepsReq{
+		respChan: respChan,
+	}:
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+
+	select {
+	case pendingSweeps := <-respChan:
+		return pendingSweeps, nil
+	case <-s.quit:
+		return nil, ErrSweeperShuttingDown
+	}
+}
+
+// handlePendingSweepsReq handles a request to retrieve all pending inputs the
+// UtxoSweeper is attempting to sweep.
+func (s *UtxoSweeper) handlePendingSweepsReq(req *pendingSweepsReq) []*PendingInput {
+	pendingInputs := make([]*PendingInput, 0, len(s.pendingInputs))
+	for _, pendingInput := range s.pendingInputs {
+		pendingInputs = append(pendingInputs, pendingInput)
```

Comment:
Should a deep copy be made here? To really get a snapshot? And also to prevent possible concurrency issues when the caller starts doing things with the inputs? (maybe just unsafe read if that is possible)

Reply:
Done.